packages/synapse-react-client/src/components/CitationPopover/CitationPopover.test.tsx

```diff
@@ -0,0 +1,194 @@
+import { act, cleanup, render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { UseQueryResult } from '@tanstack/react-query'
+import CitationPopover from './CitationPopover'
+import { useCitation } from './useCitation'
+
+jest.mock('./useCitation', () => ({
+  useCitation: jest.fn(),
+}))
+
+const mockUseCitation = useCitation as jest.MockedFunction<typeof useCitation>
+
+const openPopover = async () => {
+  const button = screen.getByTestId('CiteAsButton')
+  await act(async () => {
+    await userEvent.click(button)
+  })
+}
+
+const data =
+  '@misc{test2025,\n  title = {Some BibTeX Entry},\n  year = {2025}\n}'
+
+const mockProps = {
+  doi: 'https://doi.org/10.1234/abcd1234',
+  title: 'Some Article',
+  boilerplateText: 'Some boilerplate text',
+}
+
+describe('CitationPopover tests', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockUseCitation.mockReturnValue({
+      isLoading: false,
+      data: data,
+      error: null,
+      isError: false,
+      isSuccess: true,
+    } as UseQueryResult<string, Error>)
+  })
+  afterEach(() => {
+    cleanup()
+  })
+
+  it('renders button', () => {
+    render(<CitationPopover {...mockProps} />)
+    const button = screen.queryByTestId('CiteAsButton')
+    expect(button).toBeInTheDocument()
+  })
+
+  it('opens popover when button is clicked and fetches citation', async () => {
+    render(<CitationPopover {...mockProps} />)
+    openPopover()
+
+    await waitFor(() => {
+      expect(screen.getByTestId('CiteAsPopover')).toBeInTheDocument()
+    })
+
+    await waitFor(() => {
+      expect(mockUseCitation).toHaveBeenCalledWith(
+        mockProps.doi,
+        'bibtex',
+        true,
+      )
+    })
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(content => content.includes('Some BibTeX Entry')),
+      ).toBeInTheDocument()
+    })
```

Comment:
More succinct

```suggestion
    await screen.findByText(content => content.includes('Some BibTeX Entry'))
```

But I have never seen this pattern. What's the difference between `getByText('foo')` and `getByText(content => content.includes('foo'))`?

Reply:
Checks for any element with the substring!

Reply:
Cool, I usually use a RegEx for this but your way seems less error prone